probe-rs/src/flashing/flasher.rs

```diff
@@ -194,14 +194,24 @@ impl<'session> Flasher<'session> {
 
     pub(super) fn run_erase_all(&mut self) -> Result<(), FlashError> {
         if self.session.has_sequence_erase_all() {
+            self.progress.started_erasing();
+
             self.session
                 .sequence_erase_all()
                 .map_err(|e| FlashError::ChipEraseFailed {
                     source: Box::new(e),
                 })?;
             // We need to reload the flasher, since the debug sequence erase
             // may have invalidated any previously invalid state
-            self.load()
+            let result = self.load();
+
+            if result.is_ok() {
+                self.progress.finished_erasing();
+            } else {
+                self.progress.failed_erasing();
```

Comment:
I think if line 199 fails, then this doesn't get emitted.

Reply:
Good catch! Fixed!